src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets

```diff
@@ -58,4 +65,82 @@
   <Target Name="CopySerializerForReferenceAssemblies" AfterTargets="PrepareForPublish" Condition="@(SerializationAssembly)!=''">
     <Copy Condition="Exists('$(OutputPath)%(SerializationAssembly.Identity).XmlSerializers.dll') == 'true'" SourceFiles="$(OutputPath)%(SerializationAssembly.Identity).XmlSerializers.dll" DestinationFolder="$(PublishDir)" SkipUnchangedFiles="false" />
   </Target>
+
+  <UsingTask
+     TaskName="SgenParameters"
+     TaskFactory="RoslynCodeTaskFactory"
+     AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <Types />
+      <References />
+      <UseProxyTypes ParameterType="System.Boolean" />
+      <Verbose ParameterType="System.Boolean" />
+      <KeyFile />
+      <KeyContainer />
+      <DelaySign ParameterType="System.Boolean" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System" />
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.Text" />
+      <Using Namespace="System.Collections.Generic" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[     
+        var sgenCommands=new List<string>();
+        if(Types!=null)
+          sgenCommands.Add("--type "+ Types);
+        if(UseProxyTypes)
+          sgenCommands.Add("--proxytypes");
+        if(Verbose)
+          sgenCommands.Add("--verbose");
+        if(References!=null)
+        {
+          sgenCommands.Add("--reference "+References);
+        }
+        
+        var cscCommands=new List<string>();
+        if(KeyFile!=null)
+            cscCommands.Add("-keyfile:"+KeyFile);
+        if(KeyContainer!=null)
+            cscCommands.Add("-keycontainer:"+KeyContainer);
+        if(DelaySign)
+            cscCommands.Add("-delaysign");
+
+        if(sgenCommands.Count>0)
+        {
+            string sgenRspPath= @"$(_SgenRspFilePath)";
+            if (File.Exists(sgenRspPath))
+                File.Delete(sgenRspPath);
+            try
+            {
+                File.WriteAllLines(sgenRspPath, sgenCommands.ToArray(), Encoding.UTF8);
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine(e.Message);
+                throw;
```

Comment:
If any error throw here, will it break the compile. Sgen will only give the warning and won't break the build. The behavior should be consistent here.

Reply:
OK I'll fix it.